templates/CRM/Contribute/Form/Contribution/MainMembershipBlock.tpl

```diff
@@ -8,108 +8,71 @@
-                  <legend>{$membershipBlock.new_title}</legend>
-                {/if}
-                {if $membershipBlock.new_text}
-                  <div id="membership-intro" class="crm-section membership_new_intro-section">
-                    <p>{$membershipBlock.new_text}</p>
-                  </div>
+  <div id="priceset">
+    <fieldset>
+      {if $renewal_mode}
+        {if $membershipBlock.renewal_title}
+          <legend>{$membershipBlock.renewal_title}</legend>
+        {/if}
+        {if $membershipBlock.renewal_text}
+          <div id="membership-intro" class="crm-section membership_renewal_intro-section">
+            {$membershipBlock.renewal_text}
+          </div>
+        {/if}
+      {else}
+        {if $membershipBlock.new_title}
+          <legend>{$membershipBlock.new_title}</legend>
+        {/if}
+        {if $membershipBlock.new_text}
+          <div id="membership-intro" class="crm-section membership_new_intro-section">
+            {$membershipBlock.new_text}
+          </div>
+        {/if}
+      {/if}
+      {if !empty($membershipTypes)}
+        {foreach from=$membershipTypes item=row}
+          {if array_key_exists( 'current_membership', $row )}
+            <div class='help'>
+              {* Lifetime memberships have no end-date so current_membership array key exists but is NULL *}
+              {if $row.current_membership}
+                {if $row.current_membership|crmDate:"%Y%m%d" LT $smarty.now|crmDate:"%Y%m%d"}
+                  {ts 1=$row.current_membership|crmDate 2=$row.name}Your <strong>%2</strong> membership expired on %1.{/ts}<br />
+                {else}
+                  {ts 1=$row.current_membership|crmDate 2=$row.name}Your <strong>%2</strong> membership expires on %1.{/ts}<br />
                 {/if}
-            {/if}
-
-        </fieldset>
-      </div>
-
-    {/if}{* membership block end here *}
+              {else}
+                {ts 1=$row.name}Your <strong>%1</strong> membership does not expire (you do not need to renew that membership).{/ts}<br />
+              {/if}
+            </div>
+          {/if}
+        {/foreach}
+      {/if}
+      {include file="CRM/Price/Form/PriceSet.tpl" extends="Membership"}
+    </fieldset>
+  </div>
+  {literal}
+  <script type="text/javascript">
+    CRM.$(function($) {
+      //if price set is set we use below below code to show for showing auto renew
+      var autoRenewOption =  {/literal}'{$autoRenewOption}'{literal};
+      var autoRenew = $("#auto_renew_section");
+      var autoRenewCheckbox = $("#auto_renew");
+      var forceRenew = $("#force_renew");
+      autoRenew.hide();
+      forceRenew.hide();
+      if ( autoRenewOption == 1 ) {
+        autoRenew.show();
+      } else if ( autoRenewOption == 2 ) {
+        autoRenewCheckbox.prop('checked',  true );
+        autoRenewCheckbox.attr( 'readonly', true );
+        autoRenew.hide();
+        forceRenew.show();
+      }
+    });
+  </script>
+{/literal}
 </div>
```

Comment:
now I'm confused - I didn't intentionally remove a dif or fieldset & they still seem to match up but it shows that I have removed a couple of closers?

Reply:
hmm local diff has them inside the if

![image](https://github.com/civicrm/civicrm-core/assets/336308/7a5f860c-95b1-44d9-adca-a46d9b387fd5)

Reply:
It is possible github being github and just picking the wrong </div> to say is removed